mongo/find.go

```diff
@@ -296,6 +296,10 @@ func generateCursor(result interface{}, paginatedField string, shouldSecondarySo
 	} else {
 		typ = val.Type()
```

Comment:
You did not change this but can you get rid of this else by defining and setting `typ` like this? `typ := val.Type() `